<a href="https://colab.research.google.com/github/jeanbreno/pb-arquitetura-comp-redes-so/blob/main/Jean_Breno_AT_Andre.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **INSTITUTO INFNET**
##ALUNO: JEAN BRENO LIMA DE OLIVEIRA
##CURSO: ENGENHARIA DE SOFTWARE
###EMAIL: jean.oliveira@al.infnet.edu.br
---
FUNDAMENTOS DE DESENVOLVIMENTO PYTHON
* AT

PROFESSOR ANDRÉ VICTOR

---

---


# **Moodle**

## **1 - Tupla e ordem crescente**

In [ ]:
numeros_inteiros = (5,4,7)
print(sorted(numeros_inteiros))

[4, 5, 7]


## **2 - Soma de números pares**

In [ ]:
numero = int(input("Digite o número de termos: "))
soma = 0
n_par = []
for i in range(1, numero + 1):
    if i % 2 == 0:
      soma += i
      n_par.append(i)
print(f"A soma dos números pares até {numero} é: {soma}.")
print(f"Números pares até {numero}: {n_par}.")

Digite o número de termos: 10
A soma dos números pares até 10 é: 30.
Números pares até 10: [2, 4, 6, 8, 10].


## **3 - Potencia**

In [ ]:
def potencia(A, B):
  if (A == 0):
    return 0
  elif (B == 1):
    return A
  elif (B == 0):
    return 1
  elif (B < 0):
    exp_menor_qzero = "Expoente negativo."
    return exp_menor_qzero
  else:
    total = 1
    for i in range(B):
      total = total * A
    return total

In [ ]:
base = int(input("Digite o valor para base: "))
expoente = int(input("Digite o valor para o expoente: "))
resultado_potencia = potencia(base, expoente)
print(f"O resultado de {base} elevado ao expoente {expoente} é: {resultado_potencia}.")

Digite o valor para base: 2
Digite o valor para o expoente: 7
O resultado de 2 elevado ao expoente 7 é: 128.


In [ ]:
potencia(0,2)

0

In [ ]:
potencia(8,1)

8

In [ ]:
potencia(8,0)

1

In [ ]:
potencia(8,-1)

'Expoente negativo.'

In [ ]:
potencia(8,2)

64

## **4 - Invertendo um Vetor**

In [ ]:
import random

In [ ]:
vetor_q4 = []
for i in range(1, 6):
  elemento_q4 = random.randrange(50)
  vetor_q4.append(elemento_q4)

print(f"Lista original: {vetor_q4}")
vetor_q4.reverse()
print(f"Lista invertida: {vetor_q4}")



Lista original: [44, 38, 13, 37, 7]
Lista invertida: [7, 37, 13, 38, 44]


## **6 - Tupla pares e ímpares**

In [ ]:
import random

In [ ]:
def separador_pares_impares():
  lista_q6 = []
  lista_impares = []
  lista_pares = []
  lista_indice_pares = []
  for j in range(10):
    elemento_q6 = random.randrange(40)
    lista_q6.append(elemento_q6)
    if elemento_q6 % 2 != 0:
      lista_impares.append(elemento_q6)
    else:
      lista_pares.append(elemento_q6)
      indice_pares = lista_q6.index(elemento_q6)
      lista_indice_pares.append(indice_pares)

  resultado_q6 = f"Lista original: {lista_q6}\n"  
  resultado_q6 += f"Lista dos números ímpares: {lista_impares}\n"
  resultado_q6 += f"Lista com os números pares: {lista_pares}\n"
  resultado_q6 += f"Tupla com os índices dos números pares: {tuple(lista_indice_pares)}\n"
  print(resultado_q6)


In [ ]:
separador_pares_impares()

Lista original: [7, 9, 1, 14, 3, 35, 10, 38, 36, 1]
Lista dos números ímpares: [7, 9, 1, 3, 35, 1]
Lista com os números pares: [14, 10, 38, 36]
Tupla com os índices dos números pares: (3, 6, 7, 8)



# **Análise de Dados**


> Carregue o dataset Video_Games_Sales_as_at_22_Dec_2016.csv que contém as vendas de games em todo mundo

Dataset disponível em https://www.kaggle.com/sidtwr/videogames-sales-dataset?select=Video_Games_Sales_as_at_22_Dec_2016.csv

---




In [ ]:
import pandas as pd

In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [ ]:
import os
os.chdir("/content/drive/MyDrive/AT - Fundamentos Python")

In [ ]:
!ls -la

total 1706
-rw------- 1 root root   26654 Jun 22 03:01 Jean_Breno_AT.ipynb
-rw------- 1 root root   67304 Oct 23  2019 PS4_GamesSales.csv
-rw------- 1 root root 1610764 Jun 22 02:12 Video_Games_Sales_as_at_22_Dec_2016.csv
-rw------- 1 root root   40876 Oct 23  2019 XboxOne_GameSales.csv


In [ ]:
video_games = pd.read_csv('Video_Games_Sales_as_at_22_Dec_2016.csv')
video_games

Name Platform  ...  Developer Rating
0                         Wii Sports      Wii  ...   Nintendo      E
1                  Super Mario Bros.      NES  ...        NaN    NaN
2                     Mario Kart Wii      Wii  ...   Nintendo      E
3                  Wii Sports Resort      Wii  ...   Nintendo      E
4           Pokemon Red/Pokemon Blue       GB  ...        NaN    NaN
...                              ...      ...  ...        ...    ...
16714  Samurai Warriors: Sanada Maru      PS3  ...        NaN    NaN
16715               LMA Manager 2007     X360  ...        NaN    NaN
16716        Haitaka no Psychedelica      PSV  ...        NaN    NaN
16717               Spirits & Spells      GBA  ...        NaN    NaN
16718            Winning Post 8 2016      PSV  ...        NaN    NaN

[16719 rows x 16 columns]

## **1) Explique quantos atributos têm esse dataset e faça uma breve explicação sobre o conteúdo de cada um**


No dataset existem 16 atributos(colunas), como mostrado abaixo. 


In [ ]:
video_games.shape

(16719, 16)

In [ ]:
colunas = video_games.columns
colunas_serie = pd.Series(colunas)
colunas_serie

0                Name
1            Platform
2     Year_of_Release
3               Genre
4           Publisher
5            NA_Sales
6            EU_Sales
7            JP_Sales
8         Other_Sales
9        Global_Sales
10       Critic_Score
11       Critic_Count
12         User_Score
13         User_Count
14          Developer
15             Rating
dtype: object

* **Name:** Nome do Jogo
* **Platform:** Plataforma para que foi desenvolvido
* **Year_of_Release:** Ano de lançamento
* **Genre:** Gênero do Jogo
* **Publisher:** Quem publicou o jogo
* **NA_Sales:** Quantidade de vendas na América do Norte
* **EU_Sales:** Quantidade de vendas na Europa
* **JP_Sales:** Quantidade de vendas no Japão
* **Other_Sales:** Quantidade de vendas em outros países
* **Global_Sales:** Quantidade de vendas Global
* **Critic_Score:**
* **Critic_Count:**
* **User_Score:**
* **User_Count:**
* **Developer:** Quem desenvolveu o jogo
* **Rating:** Avaliação do jogo

## **2) Quantos games estão disponíveis no dataset ?**

In [ ]:
video_games.shape

(16719, 16)

In [ ]:
video_games['Name'].value_counts()

Need for Speed: Most Wanted                 12
Madden NFL 07                                9
Ratatouille                                  9
FIFA 14                                      9
LEGO Marvel Super Heroes                     9
                                            ..
Star Soldier: Vanishing Earth                1
Wario: Master of Disguise                    1
Grisaia no Rakuen: Le Eden de la Grisaia     1
Genma Onimusha                               1
Earth Defense Force 2017                     1
Name: Name, Length: 11562, dtype: int64

In [ ]:
video_games['Name'].value_counts().sum()

16717

De acordo com a pesquisa, 11562 jogos estão disponíveis no dataset. Pois, a diferença encontrada entre o total de linhas do atributo e o número total de games se dá pq existem jogos repetidos no dataset.

## **3) Existe algum atributo nesse dataset que não está preenchido (valores nulos) ?**

In [ ]:
video_games.isnull().sum()

Name                  2
Platform              0
Year_of_Release     269
Genre                 2
Publisher            54
NA_Sales              0
EU_Sales              0
JP_Sales              0
Other_Sales           0
Global_Sales          0
Critic_Score       8582
Critic_Count       8582
User_Score         9129
User_Count         9129
Developer          6623
Rating             6769
dtype: int64

Sim. Como mostra a pesquisa acima, existem alguns atributos que não estão 
preenchidos.


> Valores diferentes de 0



## **4) Cria 3 séries distintas de valores de vendas: uma para a America do Norte, outra para a Europa e outra para o Japão. Todas indexados pelo nome do game**

In [ ]:
video_games.groupby(['Name']).sum()['NA_Sales']

Name
 Beyblade Burst                 0.00
 Fire Emblem Fates              0.81
 Frozen: Olaf's Quest           0.48
 Haikyu!! Cross Team Match!     0.00
 Tales of Xillia 2              0.20
                                ... 
thinkSMART: Chess for Kids      0.01
uDraw Studio                    1.65
uDraw Studio: Instant Artist    0.07
wwe Smackdown vs. Raw 2006      1.57
¡Shin Chan Flipa en colores!    0.00
Name: NA_Sales, Length: 11562, dtype: float64

In [ ]:
video_games.groupby(['Name']).sum()['EU_Sales']

Name
 Beyblade Burst                 0.00
 Fire Emblem Fates              0.23
 Frozen: Olaf's Quest           0.53
 Haikyu!! Cross Team Match!     0.00
 Tales of Xillia 2              0.12
                                ... 
thinkSMART: Chess for Kids      0.00
uDraw Studio                    0.57
uDraw Studio: Instant Artist    0.10
wwe Smackdown vs. Raw 2006      1.02
¡Shin Chan Flipa en colores!    0.00
Name: EU_Sales, Length: 11562, dtype: float64

In [ ]:
video_games.groupby(['Name']).sum()['JP_Sales']

Name
 Beyblade Burst                 0.03
 Fire Emblem Fates              0.52
 Frozen: Olaf's Quest           0.00
 Haikyu!! Cross Team Match!     0.04
 Tales of Xillia 2              0.45
                                ... 
thinkSMART: Chess for Kids      0.00
uDraw Studio                    0.00
uDraw Studio: Instant Artist    0.00
wwe Smackdown vs. Raw 2006      0.00
¡Shin Chan Flipa en colores!    0.14
Name: JP_Sales, Length: 11562, dtype: float64

## **5) Converta o atributo Year_of_Release no dataframe para inteiro**

In [ ]:
video_games.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 16719 entries, 0 to 16718
Data columns (total 16 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Name             16717 non-null  object 
 1   Platform         16719 non-null  object 
 2   Year_of_Release  16450 non-null  float64
 3   Genre            16717 non-null  object 
 4   Publisher        16665 non-null  object 
 5   NA_Sales         16719 non-null  float64
 6   EU_Sales         16719 non-null  float64
 7   JP_Sales         16719 non-null  float64
 8   Other_Sales      16719 non-null  float64
 9   Global_Sales     16719 non-null  float64
 10  Critic_Score     8137 non-null   float64
 11  Critic_Count     8137 non-null   float64
 12  User_Score       7590 non-null   float64
 13  User_Count       7590 non-null   float64
 14  Developer        10096 non-null  object 
 15  Rating           9950 non-null   object 
dtypes: float64(10), object(6)
memory usage: 2.0+ MB


In [ ]:
video_games.isnull().sum()

Name                  2
Platform              0
Year_of_Release     269
Genre                 2
Publisher            54
NA_Sales              0
EU_Sales              0
JP_Sales              0
Other_Sales           0
Global_Sales          0
Critic_Score       8582
Critic_Count       8582
User_Score         9129
User_Count         9129
Developer          6623
Rating             6769
dtype: int64

In [ ]:
ano_lan = video_games['Year_of_Release']

In [ ]:
ano_lan_D=ano_lan.dropna(axis=0)

In [ ]:
ano_lan_D.astype(int)

0        2006
1        1985
2        2008
3        2009
4        1996
         ... 
16714    2016
16715    2006
16716    2016
16717    2003
16718    2016
Name: Year_of_Release, Length: 16450, dtype: int64

## **6) Remova as linhas que tenham algum atributo nulo entre os atributos de vendas pelo mundo (considere apenas NA, EU e JP)**

In [ ]:
video_games.isnull().sum()

Name                  2
Platform              0
Year_of_Release     269
Genre                 2
Publisher            54
NA_Sales              0
EU_Sales              0
JP_Sales              0
Other_Sales           0
Global_Sales          0
Critic_Score       8582
Critic_Count       8582
User_Score         9129
User_Count         9129
Developer          6623
Rating             6769
dtype: int64

De acordo com a pesquisa, não há nenhum valor nulo nos atributos NA, EU e JP.

## **7) Quantos valores distintos existem para cada série ?**

In [ ]:
video_games.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 16719 entries, 0 to 16718
Data columns (total 16 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Name             16717 non-null  object 
 1   Platform         16719 non-null  object 
 2   Year_of_Release  16450 non-null  float64
 3   Genre            16717 non-null  object 
 4   Publisher        16665 non-null  object 
 5   NA_Sales         16719 non-null  float64
 6   EU_Sales         16719 non-null  float64
 7   JP_Sales         16719 non-null  float64
 8   Other_Sales      16719 non-null  float64
 9   Global_Sales     16719 non-null  float64
 10  Critic_Score     8137 non-null   float64
 11  Critic_Count     8137 non-null   float64
 12  User_Score       7590 non-null   float64
 13  User_Count       7590 non-null   float64
 14  Developer        10096 non-null  object 
 15  Rating           9950 non-null   object 
dtypes: float64(10), object(6)
memory usage: 2.0+ MB


## **8) Qual o game mais vendido na América do Norte ?**

In [ ]:
mais_vendidos_NA = pd.read_csv('Video_Games_Sales_as_at_22_Dec_2016.csv', usecols=['Name', 'NA_Sales'], index_col=['Name'], squeeze=True)
mais_vendidos_NA.sort_values(ascending=False)

Name
Wii Sports                                                         41.36
Super Mario Bros.                                                  29.08
Duck Hunt                                                          26.93
Tetris                                                             23.20
Mario Kart Wii                                                     15.68
                                                                   ...  
Toukiden 2                                                          0.00
SD Gundam G Generation Genesis                                      0.00
Pachitte Chonmage Tatsujin 13: Pachinko Hissatsu Shigotojin III     0.00
Theme Park                                                          0.00
Bomberman B-Daman                                                   0.00
Name: NA_Sales, Length: 16719, dtype: float64

O game mais vendido na América do Norte é o Wii Sports.

## **9) Qual o game mais vendido na Europa ?**

In [ ]:
mais_vendidos_EU = pd.read_csv('Video_Games_Sales_as_at_22_Dec_2016.csv', usecols=['Name', 'EU_Sales'], index_col=['Name'], squeeze=True)
mais_vendidos_EU.sort_values(ascending=False)

Name
Wii Sports                                      28.96
Mario Kart Wii                                  12.76
Nintendogs                                      10.95
Wii Sports Resort                               10.93
Brain Age: Train Your Brain in Minutes a Day     9.20
                                                ...  
Draglade                                         0.00
All Kamen Rider: Rider Generation 2              0.00
Darius Gaiden                                    0.00
KORG DS-10 Synthesizer Plus                      0.00
Bomberman B-Daman                                0.00
Name: EU_Sales, Length: 16719, dtype: float64

O game mais vendido na Europa é o Wii Sports.

## **10) Qual o game mais vendido no Japão ?**

In [ ]:
mais_vendidos_JP = pd.read_csv('Video_Games_Sales_as_at_22_Dec_2016.csv', usecols=['Name', 'JP_Sales'], index_col=['Name'], squeeze=True)
mais_vendidos_JP.sort_values(ascending=False)

Name
Pokemon Red/Pokemon Blue         10.22
Pokemon Gold/Pokemon Silver       7.20
Super Mario Bros.                 6.81
New Super Mario Bros.             6.50
Pokemon Diamond/Pokemon Pearl     6.04
                                 ...  
Mega Man & Bass                   0.00
Airblade                          0.00
Again                             0.00
Grand Slam Tennis 2               0.00
Board Game: Top Shop              0.00
Name: JP_Sales, Length: 16719, dtype: float64

O game mais vendido no Japão é o Pokemon Red/Pokemon Blue.

## **11) Qual das 3 localidades (America do Norte, Europa e Japão) tem a maior média de games vendida ?**

In [ ]:
mais_vendidos_NA.mean()

0.07760212931394815

In [ ]:
mais_vendidos_EU.mean()

0.14502482205876838

In [ ]:
mais_vendidos_JP.mean()

0.07760212931394815

A Europa tem a maior média de jogos vendidos.

## **12) Qual o game menos vendido no mundo em números absolutos ?**

In [ ]:
menos_vendidos_GLOBAL = pd.read_csv('Video_Games_Sales_as_at_22_Dec_2016.csv', usecols=['Name', 'Global_Sales'], index_col=['Name'], squeeze=True)
menos_vendidos_GLOBAL.sort_values(ascending=True)

Name
Winning Post 8 2016                                            0.01
AKB1/48: Idol to Guam de Koishitara...                         0.01
Casper                                                         0.01
Simple 2000 Series Vol. 120: The Saigo no Nippon Tsuwamono     0.01
Rave Master: Special Attack Force!                             0.01
                                                              ...  
Pokemon Red/Pokemon Blue                                      31.37
Wii Sports Resort                                             32.77
Mario Kart Wii                                                35.52
Super Mario Bros.                                             40.24
Wii Sports                                                    82.53
Name: Global_Sales, Length: 16719, dtype: float64

O game menos vendido no mundo é o Winning Post 8 2016.

## **13) Separe numa série a parte os 5% de games mais vendidos na NA. Compare a semelhança e diferença com as outras duas localidades**

In [ ]:
def percentual(tamanho_series):
  tam = tamanho_series
  porcent = (tam * 5) / 100
  return porcent


In [ ]:
serie_mais_vendidos_NA = pd.read_csv('Video_Games_Sales_as_at_22_Dec_2016.csv', usecols=['Name', 'NA_Sales'], index_col=['Name'], squeeze=True)
serie_mais_vendidos_NA.sort_values(ascending=False)

Name
Wii Sports                                                         41.36
Super Mario Bros.                                                  29.08
Duck Hunt                                                          26.93
Tetris                                                             23.20
Mario Kart Wii                                                     15.68
                                                                   ...  
Toukiden 2                                                          0.00
SD Gundam G Generation Genesis                                      0.00
Pachitte Chonmage Tatsujin 13: Pachinko Hissatsu Shigotojin III     0.00
Theme Park                                                          0.00
Bomberman B-Daman                                                   0.00
Name: NA_Sales, Length: 16719, dtype: float64

In [ ]:
tamanho_da_serie = len(serie_mais_vendidos_NA)
#chamada da função
percent = percentual(tamanho_da_serie)
percent

835.95

In [ ]:
total_5p_NA = serie_mais_vendidos_NA.iloc[0 : round(percent)].sum()

In [ ]:
serie_mais_vendidos_EU = pd.read_csv('Video_Games_Sales_as_at_22_Dec_2016.csv', usecols=['Name', 'EU_Sales'], index_col=['Name'], squeeze=True)
serie_mais_vendidos_EU.sort_values(ascending=False)

Name
Wii Sports                                      28.96
Mario Kart Wii                                  12.76
Nintendogs                                      10.95
Wii Sports Resort                               10.93
Brain Age: Train Your Brain in Minutes a Day     9.20
                                                ...  
Draglade                                         0.00
All Kamen Rider: Rider Generation 2              0.00
Darius Gaiden                                    0.00
KORG DS-10 Synthesizer Plus                      0.00
Bomberman B-Daman                                0.00
Name: EU_Sales, Length: 16719, dtype: float64

In [ ]:
total_5p_EU = serie_mais_vendidos_EU.iloc[0 : round(percent)].sum()

In [ ]:
serie_mais_vendidos_JP = pd.read_csv('Video_Games_Sales_as_at_22_Dec_2016.csv', usecols=['Name', 'JP_Sales'], index_col=['Name'], squeeze=True)
serie_mais_vendidos_JP.sort_values(ascending=False)

Name
Pokemon Red/Pokemon Blue         10.22
Pokemon Gold/Pokemon Silver       7.20
Super Mario Bros.                 6.81
New Super Mario Bros.             6.50
Pokemon Diamond/Pokemon Pearl     6.04
                                 ...  
Mega Man & Bass                   0.00
Airblade                          0.00
Again                             0.00
Grand Slam Tennis 2               0.00
Board Game: Top Shop              0.00
Name: JP_Sales, Length: 16719, dtype: float64

In [ ]:
total_5p_JP = serie_mais_vendidos_JP.iloc[0 : round(percent)].sum()

In [ ]:
print(f"5% dos jogos vendidos na NA: {round(total_5p_NA)} jogos")
print(f"5% dos jogos vendidos na EU: {round(total_5p_EU)} jogos")
print(f"5% dos jogos vendidos no JP: {round(total_5p_JP)} jogos")

5% dos jogos vendidos na NA: 1928 jogos
5% dos jogos vendidos na EU: 1172 jogos
5% dos jogos vendidos no JP: 513 jogos


## **14) Qual a plataforma de games com maior número de games neste dataset?**

In [ ]:
plataformas = pd.read_csv('Video_Games_Sales_as_at_22_Dec_2016.csv', usecols=['Name', 'Platform'], index_col=['Platform'], squeeze=True)
plataformas

Platform
Wii                        Wii Sports
NES                 Super Mario Bros.
Wii                    Mario Kart Wii
Wii                 Wii Sports Resort
GB           Pokemon Red/Pokemon Blue
                    ...              
PS3     Samurai Warriors: Sanada Maru
X360                 LMA Manager 2007
PSV           Haitaka no Psychedelica
GBA                  Spirits & Spells
PSV               Winning Post 8 2016
Name: Name, Length: 16719, dtype: object

In [ ]:
plataformas.loc['Wii']

Platform
Wii                                           Wii Sports
Wii                                       Mario Kart Wii
Wii                                    Wii Sports Resort
Wii                                             Wii Play
Wii                            New Super Mario Bros. Wii
                             ...                        
Wii                                            Bomberman
Wii                     The Ultimate Battle of the Sexes
Wii             Mahou Sensei Negima!? Neo-Pactio Fight!!
Wii                Help Wanted: 50 Wacky Jobs (jp sales)
Wii    Yattaman Wii: BikkuriDokkiri Machine de Mou Ra...
Name: Name, Length: 1320, dtype: object

In [ ]:
plataformas.loc['PS2']

Platform
PS2                       Grand Theft Auto: San Andreas
PS2                         Grand Theft Auto: Vice City
PS2                              Gran Turismo 3: A-Spec
PS2                                Grand Theft Auto III
PS2                                      Gran Turismo 4
                             ...                       
PS2        Sugar + Spice! Anoko no Suteki na Nanimokamo
PS2                                      Kanokon: Esuii
PS2                    Scarlett: Nichijou no Kyoukaisen
PS2                                           Real Rode
PS2    SCORE International Baja 1000: The Official Game
Name: Name, Length: 2161, dtype: object

In [ ]:
plataformas.loc['X360']

Platform
X360                 Kinect Adventures!
X360                 Grand Theft Auto V
X360     Call of Duty: Modern Warfare 3
X360            Call of Duty: Black Ops
X360         Call of Duty: Black Ops II
                     ...               
X360           Dragon Ball Z for Kinect
X360    World Snooker Championship 2007
X360        Metal Gear Solid HD Edition
X360                 Ben 10 Omniverse 2
X360                   LMA Manager 2007
Name: Name, Length: 1262, dtype: object

## **15) Verifique se as plataformas com maior total de vendas em cada região são as mesmas**

In [ ]:
video_games.groupby(['Platform']).sum()['NA_Sales']

Platform
2600     90.60
3DO       0.00
3DS      83.49
DC        5.43
DS      382.67
GB      114.32
GBA     187.54
GC      133.46
GEN      21.05
GG        0.00
N64     139.02
NES     125.94
NG        0.00
PC       94.53
PCFX      0.00
PS      336.52
PS2     583.84
PS3     393.49
PS4     108.74
PSP     109.17
PSV      12.58
SAT       0.72
SCD       1.00
SNES     61.23
TG16      0.00
WS        0.00
Wii     496.90
WiiU     38.10
X360    602.47
XB      186.69
XOne     93.12
Name: NA_Sales, dtype: float64

In [ ]:
video_games.groupby(['Platform']).sum()['EU_Sales']

Platform
2600      5.47
3DO       0.00
3DS      61.48
DC        1.69
DS      188.89
GB       47.82
GBA      75.25
GC       38.71
GEN       6.05
GG        0.00
N64      41.06
NES      21.15
NG        0.00
PC      142.44
PCFX      0.00
PS      213.61
PS2     339.29
PS3     330.29
PS4     141.09
PSP      66.68
PSV      13.12
SAT       0.54
SCD       0.36
SNES     19.04
TG16      0.00
WS        0.00
Wii     262.21
WiiU     25.13
X360    270.76
XB       60.95
XOne     51.59
Name: EU_Sales, dtype: float64

In [ ]:
video_games.groupby(['Platform']).sum()['JP_Sales']

Platform
2600      0.00
3DO       0.10
3DS     100.67
DC        8.56
DS      175.57
GB       85.12
GBA      47.33
GC       21.58
GEN       2.70
GG        0.04
N64      34.22
NES      98.65
NG        1.44
PC        0.17
PCFX      0.03
PS      139.82
PS2     139.20
PS3      80.19
PS4      16.00
PSP      76.78
PSV      21.93
SAT      32.26
SCD       0.45
SNES    116.55
TG16      0.16
WS        1.42
Wii      69.33
WiiU     13.01
X360     12.43
XB        1.38
XOne      0.34
Name: JP_Sales, dtype: float64

In [ ]:
video_games.groupby(['Platform']).sum()['Other_Sales']

Platform
2600      0.91
3DO       0.00
3DS      13.36
DC        0.27
DS       59.28
GB        8.20
GBA       7.73
GC        5.18
GEN       0.97
GG        0.00
N64       4.38
NES       5.31
NG        0.00
PC       22.38
PCFX      0.00
PS       40.91
PS2     193.44
PS3     135.68
PS4      48.35
PSP      41.42
PSV       6.46
SAT       0.07
SCD       0.05
SNES      3.22
TG16      0.00
WS        0.00
Wii      79.07
WiiU      5.95
X360     85.76
XB        8.72
XOne     14.27
Name: Other_Sales, dtype: float64

In [ ]:
video_games.groupby(['Platform']).sum()['Global_Sales']

Platform
2600      97.08
3DO        0.10
3DS      259.09
DC        15.97
DS       807.10
GB       255.45
GBA      318.50
GC       199.36
GEN       30.78
GG         0.04
N64      218.88
NES      251.07
NG         1.44
PC       260.30
PCFX       0.03
PS       730.68
PS2     1255.64
PS3      939.43
PS4      314.23
PSP      294.30
PSV       54.12
SAT       33.59
SCD        1.87
SNES     200.05
TG16       0.16
WS         1.42
Wii      908.13
WiiU      82.16
X360     971.63
XB       258.26
XOne     159.44
Name: Global_Sales, dtype: float64

## **16) Verifique qual o total de vendas de jogos de Sports no Xbox na América do Norte**

In [ ]:
serie_game_xbox = pd.read_csv('Video_Games_Sales_as_at_22_Dec_2016.csv', usecols=['Name', 'Platform', 'Genre', 'NA_Sales'], index_col=['Genre'])
serie_game_xbox

Name Platform  NA_Sales
Genre                                                         
Sports                           Wii Sports      Wii     41.36
Platform                  Super Mario Bros.      NES     29.08
Racing                       Mario Kart Wii      Wii     15.68
Sports                    Wii Sports Resort      Wii     15.61
Role-Playing       Pokemon Red/Pokemon Blue       GB     11.27
...                                     ...      ...       ...
Action        Samurai Warriors: Sanada Maru      PS3      0.00
Sports                     LMA Manager 2007     X360      0.00
Adventure           Haitaka no Psychedelica      PSV      0.00
Platform                   Spirits & Spells      GBA      0.01
Simulation              Winning Post 8 2016      PSV      0.00

[16719 rows x 3 columns]

In [ ]:
serie_game_xbox.groupby(['Genre', 'Platform']).sum()

NA_Sales
Genre    Platform          
Action   2600         27.39
         3DS          19.80
         DC            0.41
         DS           65.00
         GB            4.07
...                     ...
Strategy Wii           2.32
         WiiU          0.49
         X360          6.50
         XB            2.05
         XOne          0.27

[293 rows x 1 columns]

In [ ]:
df_X360 = serie_game_xbox[serie_game_xbox['Platform'] =='X360']
df_XB = serie_game_xbox[serie_game_xbox['Platform'] =='XB']
df_XOne = serie_game_xbox[serie_game_xbox['Platform'] =='XOne']

In [ ]:
lista_X=[df_X360, df_XB, df_XOne]
lista_X

[                                     Name Platform  NA_Sales
 Genre                                                       
 Misc                   Kinect Adventures!     X360     15.00
 Action                 Grand Theft Auto V     X360      9.66
 Shooter    Call of Duty: Modern Warfare 3     X360      9.04
 Shooter           Call of Duty: Black Ops     X360      9.70
 Shooter        Call of Duty: Black Ops II     X360      8.25
 ...                                   ...      ...       ...
 Fighting         Dragon Ball Z for Kinect     X360      0.01
 Sports    World Snooker Championship 2007     X360      0.00
 Action        Metal Gear Solid HD Edition     X360      0.00
 Action                 Ben 10 Omniverse 2     X360      0.00
 Sports                   LMA Manager 2007     X360      0.00
 
 [1262 rows x 3 columns],
                                           Name Platform  NA_Sales
 Genre                                                            
 Shooter                        

In [ ]:
df_X360_result = df_X360.groupby(['Genre']).sum()
df_X360_result.loc[['Sports']]

NA_Sales
Genre           
Sports     90.02

**O total de vendas na NA de jogos de esportes da plataforma X360 foi de 90.02.**


In [ ]:
df_XB_result = df_XB.groupby(['Genre']).sum()
df_XB_result.loc[['Sports']]

NA_Sales
Genre           
Sports     41.02

**O total de vendas na NA de jogos de esportes da plataforma XB foi de 41.02.**

In [ ]:
df_XOne_result = df_XOne.groupby(['Genre']).sum()
df_XOne_result.loc[['Sports']]

NA_Sales
Genre           
Sports     15.59

**O total de vendas na NA de jogos de esportes da plataforma XOne foi de 15.59.**